Civi/Api4/Event/Subscriber/ContactSchemaMapSubscriber.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Civi\Api4\Event\Subscriber;
+
+use Civi\Api4\Event\Events;
+use Civi\Api4\Event\SchemaMapBuildEvent;
+use Civi\Api4\Service\Schema\Joinable\Joinable;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ContactSchemaMapSubscriber implements EventSubscriberInterface {
+
+  /**
+   * @return array
+   */
+  public static function getSubscribedEvents() {
+    return [
+      Events::SCHEMA_MAP_BUILD => 'onSchemaBuild',
+    ];
+  }
+
+  /**
+   * @param \Civi\Api4\Event\SchemaMapBuildEvent $event
+   */
+  public function onSchemaBuild(SchemaMapBuildEvent $event) {
+    $schema = $event->getSchemaMap();
+    $table = $schema->getTableByName('civicrm_contact');
+
+    // Add links to primary email, address, phone & im
```

Comment:
There is ALSO `openid` - that is the bane of my life. I would be happy to see us not cater to that OR `im` which I really doubt people use - but I could be wrong - but completeness would include both

Reply:
People do use IM, although that term has a bit of a retro feel, there are still plenty of uses for it. civicrm.org uses it to record a person's mattermost username.